site2/docs/security-encryption.md

```diff
@@ -78,18 +78,20 @@ class RawFileKeyReader implements CryptoKeyReader {
         return keyInfo;
     }
 }
-PulsarClient pulsarClient = PulsarClient.create("http://localhost:8080");
 
-ProducerConfiguration prodConf = new ProducerConfiguration();
-prodConf.setCryptoKeyReader(new RawFileKeyReader("test_ecdsa_pubkey.pem", "test_ecdsa_privkey.pem"));
-prodConf.addEncryptionKey("myappkey");
+PulsarClient pulsarClient = PulsarClient.builder().serviceUrl("pulsar://localhost:6650").build();
 
-Producer producer = pulsarClient.createProducer("persistent://my-tenant/my-ns/my-topic", prodConf);
+Producer producer = pulsarClient.newProducer()
+                .topic("persistent://my-tenant/my-ns/my-topic")
+                .addEncryptionKey("myappkey")
```

Comment:
what will be the myappkey

Reply:
Its just a “tag” for the key. So the consumer knows with which key to decrypt. As basically every producer can use a different encryption key

Reply:
I used secured  AES key generated from this method
	Key key;
		SecureRandom rand = new SecureRandom();
		KeyGenerator generator = KeyGenerator.getInstance("AES");
		generator.init(256, rand);
		key = generator.generateKey();
and  created a producer
Producer<byte[]> producer = client.newProducer()
					.topic("persistent://my-tenant/my-namespace/my-topic-1")
					.cryptoKeyReader(new RawFileKeyReader("src/main/resources/test_ecdsa_pubkey.pem", "src/main/resources/test_ecdsa_privkey.pem"))
					.addEncryptionKey("key")
					.enableBatching(false)
					.create();

But i got this exception
org.apache.pulsar.client.api.PulsarClientException: java.util.concurrent.ExecutionException: org.apache.pulsar.client.api.PulsarClientException: java.lang.NoSuchMethodError: 'org.apache.pulsar.shade.io.netty.buffer.ByteBuf org.apache.pulsar.client.api.MessageCrypto.encrypt(java.util.Set, org.apache.pulsar.client.api.CryptoKeyReader, java.util.function.Supplier, org.apache.pulsar.shade.io.netty.buffer.ByteBuf)'

------------------------------------------------------------------------------------------------------------------
I followed the document

Reply:
Yep, this seems to be completely broken in 2.5.1. Works fine in 2.5.0.

Reply:
Well, it works fine if you add this dependency:
compile group: 'org.apache.pulsar', name: 'pulsar-client-messagecrypto-bc', version: '2.5.1'

Don't know if this a bug or this is how it is supposed to be now with the whole addition of bcfips support.

Reply:
I have already added that dependency 
however i have raised a ticket for this 

ISSUE-6834: Apache pulsar Encryption failed using ecdsa